crypto/kyber/p_kyber_test.cc

```diff
@@ -4,280 +4,463 @@
 #include <openssl/evp.h>
```

Comment:
Is it possible to make this the generic kem_test.cc that can test any kem's basic functionality (key generation, encap/decaps consistency) and kats?